app/frontend/src/store/modules/form.js

```diff
@@ -0,0 +1,104 @@
+import formService from '@/services/formService.js';
+
+export default {
+  namespaced: true,
+  state: {
+    form: {
+      description: '',
+      id: '',
+      idps: [],
+      name: '',
+    },
+    formList: [],
+    formSubmission: {
+      confirmationId: '',
+      submission: {
+        data: {}
+      }
+    },
+    submissionList: [],
+    version: {}
+  },
+  getters: {
+    form: state => state.form,
+    formList: state => state.formList,
+    formSubmission: state => state.formSubmission,
+    submissionList: state => state.submissionList,
+    version: state => state.version
+  },
+  mutations: {
+    ADD_FORM_TO_LIST(state, form) {
+      state.formList.push(form);
+    },
+    ADD_SUBMISSION_TO_LIST(state, submission) {
+      state.submissions.push(submission);
+    },
+    SET_FORM(state, form) {
+      state.form = form;
+    },
+    SET_FORMLIST(state, forms) {
+      state.formList = forms;
+    },
+    SET_FORMSUBMISSION(state, submission) {
+      state.formSubmission = submission;
+    },
+    SET_SUBMISSIONLIST(state, submissions) {
+      state.submissionList = submissions;
+    },
+    SET_VERSION(state, version) {
+      state.version = version;
+    },
+  },
+  actions: {
```

Comment:
As an extension to my previous comment - one of the thoughts I had for potential offline support would be to make all of the state actions be pushed into a memory-managed queue - so instead of directly just fetching the form and acting on state immediately, the action would first enqueue the action, and then the queue itself would then manage the state and update as needed. This would afford us the ability to have more fail-safe behavior in the event network traffic isn't going through (trying to submit in terrible network environment or poor cell service for example).
Probably no action at this time, but it is something we should consider if it's on the roadmap.

Reply:
👍